src/aggregate.d

```diff
@@ -215,78 +208,136 @@ public:
             sd.semanticTypeInfoMembers();
     }
 
-    abstract void finalizeSize();
-
-    override final uint size(Loc loc)
+    /***************************************
+     * Find all instance fields, then push them into `fields`.
```

Comment:
What happens if running semantic() causes more fields to be added? (Like if there are fields that are tuples.)